internal/aws/ec2.go

```diff
@@ -164,6 +167,34 @@ func (c *Cloud) GetSubnetsByNameOrID(ctx context.Context, nameOrIDs []string) (s
 	return
 }
 
+func (c *Cloud) GetClusterSubnets(tagSubnetType string) ([]*ec2.Subnet, error) {
+	var result []*ec2.Subnet
+
+	in := &ec2.DescribeSubnetsInput{Filters: []*ec2.Filter{
+		{
+			Name:   aws.String("tag:kubernetes.io/cluster/" + c.clusterName),
+			Values: aws.StringSlice([]string{"owned", "shared"}),
+		},
+	}}
+
+	subnets, err := c.describeSubnetsHelper(in)
+	if err != nil {
+		return nil, err
+	}
+
+	if subnets == nil {
+		return nil, nil
+	}
+
+	for _, s := range subnets {
+		if c.hasClusterTagHelper(s.Tags, tagSubnetType) {
```

Comment:
I think if we put this as a tag:filter alone with existing `tag:kubernetes.io/cluster/` filter, it will be more clear and simple :D